PersonalizeIDPretrained.yaml

```diff
@@ -0,0 +1,379 @@
+              Action:
+                - "s3:*"
+              Resource:
+                - !Sub arn:aws:s3:::${MLOpsStack.Outputs.PersonalizeBucketName}
+                - !Sub arn:aws:s3:::${MLOpsStack.Outputs.PersonalizeBucketName}/*
+            -
+              Effect: Allow
+              Action:
+                - "s3:GetObject*"
+              Resource:
+                - !Sub arn:aws:s3:::personalize-solution-staging-us-east-1/personalize-immersionday-media
+                - !Sub arn:aws:s3:::personalize-solution-staging-us-east-1/personalize-immersionday-media/*
+                - !Sub arn:aws:s3:::personalize-solution-staging-us-east-1/personalize-immersionday-media/imdb/*
+            -
+              Effect: Allow
+              Action:
+                - "s3:GetObject*"
+              Resource:
+                - !Sub arn:aws:s3:::retail-demo-store-us-east-1
+                - !Sub arn:aws:s3:::retail-demo-store-us-east-1/*
+            - 
+              Effect: Allow
+              Action:
+                - "bedrock:*"
+              Resource:
+                - "*"
+        Roles: 
+          - !Ref SageMakerIamRole
+          - !Ref PersonalizeIamRole
+          - !Ref BucketJanitorRole
+
+  # Amazon S3 - S3 Bucket Policy
+  # ---
+  # This policy allows Personalize to perform any actions on the 
+  # S3 bucket created by this template
+  PersonalizeBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !GetAtt MLOpsStack.Outputs.PersonalizeBucketName
+      PolicyDocument:
+        Version: '2012-10-17'
+        Id: 'PersonalizeS3BucketAccessPolicy'
+        Statement:
+          - Action:
+              - 's3:GetObject'
+              - 's3:PutObject'
+              - 's3:ListBucket'
+            Effect: Allow
+            Resource:
+              - !Sub arn:aws:s3:::${MLOpsStack.Outputs.PersonalizeBucketName}
+              - !Sub arn:aws:s3:::${MLOpsStack.Outputs.PersonalizeBucketName}/*
+            Principal:
+              Service: personalize.amazonaws.com
+              
+  SageMakerIamRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          -
+            Effect: Allow
+            Principal:
+              Service: sagemaker.amazonaws.com
+            Action: sts:AssumeRole
+      Path: "/"
+      Policies:
+        - PolicyName: !Sub "${AWS::StackName}-sagemaker-s3-poicy"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Action:
+                  - "s3:*"
+                  - "s3-object-lambda:*"
+                Resource: "*"
+                Effect: Allow
+      ManagedPolicyArns:
+        - "arn:aws:iam::aws:policy/AmazonSageMakerFullAccess"
+        - "arn:aws:iam::aws:policy/service-role/AmazonPersonalizeFullAccess"
+        - "arn:aws:iam::aws:policy/AmazonSSMReadOnlyAccess"
+        - "arn:aws:iam::aws:policy/AWSStepFunctionsFullAccess"
```

Comment:
Do we still need this policy? Guessing it was originally needed for the old MLOps workflow.

Reply:
The workshop still uses https://aws.amazon.com/solutions/implementations/maintaining-personalized-experiences-with-ml/ which I believe requires it

Reply:
That solution has its own role. This role is for the notebook which no longer accesses SF directly.